src/utility.ts

```diff
@@ -23,11 +24,19 @@ export const Utility = {
 		return new TextDecoder().decode(input);
 	},
 
-	filterToString<T>(filter: SelectorFilter<T> | LogicalFilter<T>): string {
+	filterToString<T>(filter: SelectorFilter<T> | LogicalFilter<T> | Selector<T>): string {
 		// eslint-disable-next-line no-prototype-builtins
-		return filter.hasOwnProperty("logicalOperator")
-			? Utility._logicalFilterToString(filter as LogicalFilter<T>)
-			: this.objToJsonString(this._selectorFilterToJSONObj(filter as SelectorFilter<T>));
+		if (filter.hasOwnProperty('op') && (filter['op'] === '$and' || filter['op'] === '$or')) {
```

Comment:
should we define these constants somewhere `$or`, `$and`?

Reply:
updated it to refer to enums.